### PR TITLE
Add dapr resiliency specs to container apps

### PR DESCRIPTION
**Pull request by addjuarez**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/ManagedEnvironmentsDaprResiliencyCircuitBreakerPolicies.json [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+            }
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {},
```
Comment (rkmanda) [inline specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/ManagedEnvironmentsDaprResiliencyCircuitBreakerPolicies.json:57]: >examples [](http://example.com/codeflow?start=14&length=8)

Add examples everywhere

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json [hunk 2]
```diff
@@ -747,6 +747,143 @@
...
+          }
+        },
+        "nextLink": {
+          "description": "Link to next page of resources.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "DaprResiliencyCircuitBreakerPolicy": {
```
Comment (rkmanda) [inline specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json:818]: >DaprResiliencyCircuitBreakerPolicy [](http://example.com/codeflow?start=5&length=34)

needs ARM envelope atleast as a proxy resource

Comment (rkmanda) [reply specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json:818]: do an allOf ProxyResource from the v4 common types

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json [hunk 3]
```diff
@@ -747,6 +747,143 @@
...
+      "description": "Resiliency Policy Breakers Policy",
+      "type": "object",
+      "properties": {
+        "maxRequests": {
```
Comment (rkmanda) [inline specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json:822]: >maxRequests [](http://example.com/codeflow?start=9&length=11)

Is there a min. max for this?

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json [hunk 4]
```diff
@@ -747,6 +747,143 @@
...
+          "description": "maximum number of requests allowed to pass through",
+          "type": "string"
```
Comment (rkmanda) [inline specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json:824]: >string [](http://example.com/codeflow?start=19&length=6)

integer?

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json [hunk 5]
```diff
@@ -747,6 +747,143 @@
...
+        },
+        "interval": {
+          "description": "cyclical period of time used by the CB to clear its internal counts",
+          "type": "string"
```
Comment (rkmanda) [inline specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json:828]: >": [](http://example.com/codeflow?start=15&length=2)

format timespan?

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json [hunk 6]
```diff
@@ -747,6 +747,143 @@
...
+        },
+        "timeout": {
+          "description": "the period of the open state until the CB switches to half-open",
+          "type": "string"
```
Comment (rkmanda) [inline specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json:832]: >string [](http://example.com/codeflow?start=19&length=6)

integer or timespan?

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json [hunk 7]
```diff
@@ -747,6 +747,143 @@
...
+          "description": "A Common Expression Language (CEL) statement that is evaluated by the CB When the statement evaluates to true, the CB trips and becomes open",
+          "type": "string"
+        },
+        "circuitBreakerScope": {
+          "description": "Specify whether circuit breaking state should be scoped to an individual actor ID, all actors across the actor type, or both. Possible values include id, type, or both",
+          "type": "string"
+        },
+        "circuitBreakerCacheSize": {
+          "description": "Specify a cache size for the number of CBs to keep in memory. The value should be larger than the expected number of active actor instances.",
+          "type": "string"
```
Comment (rkmanda) [inline specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json:844]: >string [](http://example.com/codeflow?start=19&length=6)

integer?

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json [hunk 8]
```diff
@@ -747,6 +747,143 @@
...
+          }
+        },
+        "nextLink": {
+          "description": "Link to next page of resources.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "DaprResiliencyTimeoutPolicy": {
```
Comment (rkmanda) [inline specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json:869]: >DaprResiliencyTimeoutPolicy [](http://example.com/codeflow?start=5&length=27)

needs the ARM envelope. same as the above resource.

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json [hunk 9]
```diff
@@ -747,6 +747,143 @@
...
+      "description": "Resiliency Policy Retries",
+      "type": "object",
+      "properties": {
+        "general": {
+          "description": "timeout duration for general operations",
+          "type": "string"
```
Comment (rkmanda) [inline specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json:875]: >string [](http://example.com/codeflow?start=19&length=6)

fix all datatypes. Everything here is a string!

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/ManagedEnvironmentsDaprResiliencyCircuitBreakerPolicies.json [hunk 10]
```diff
@@ -0,0 +1,238 @@
...
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": false,
+        "x-ms-examples": {}
```
Comment (rkmanda) [inline specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/ManagedEnvironmentsDaprResiliencyCircuitBreakerPolicies.json:167]: the default is false so no need to specify this

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/ManagedEnvironmentsDaprResiliencyCircuitBreakerPolicies.json [hunk 11]
```diff
@@ -0,0 +1,238 @@
...
+      },
+      "delete": {
+        "tags": [
+          "DaprResiliencyCircuitBreakerPolicies"
+        ],
+        "summary": "Delete a Dapr Component.",
+        "description": "Delete a Dapr Component from a Managed Environment.",
```
Comment (rkmanda) [inline specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/ManagedEnvironmentsDaprResiliencyCircuitBreakerPolicies.json:174]: >Dapr [](http://example.com/codeflow?start=33&length=4)

Expand the term Dapr atleast once at the top of the file for context

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/ManagedEnvironments.json [hunk 12]
```diff
@@ -898,6 +922,10 @@
               "description": "Vnet configuration for the environment",
               "$ref": "#/definitions/VnetConfiguration"
             },
+            "daprResiliencyDefaults": {
+              "description": "Dapr Resiliency Default Policies",
+              "$ref": "#/definitions/DaprResiliencyDefaults"
```
Comment (rkmanda) [inline specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/ManagedEnvironments.json:927]: >DaprResiliencyDefaults [](http://example.com/codeflow?start=37&length=22)

its not clear why the defaults are being published here. Can u provide an explanation for whats being achieved here?

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/ManagedEnvironmentsDaprResiliencyCircuitBreakerPolicies.json (hunk 11 above)
Comment (addjuarez) [reply specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/ManagedEnvironmentsDaprResiliencyCircuitBreakerPolicies.json]: This is the context, https://docs.dapr.io/operations/resiliency/policies/

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json (hunk 6 above)
Comment (addjuarez) [reply specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json]: timespan

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json (hunk 5 above)
Comment (addjuarez) [reply specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json]: timespan

**Comment by addjuarez** [comment]
I do not have permission to remove the `ARMChangesRequested` label

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/ManagedEnvironmentsDaprResiliencyCircuitBreakerPolicies.json (hunk 1 above)
Comment (addjuarez) [reply specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/ManagedEnvironmentsDaprResiliencyCircuitBreakerPolicies.json]: added

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json (hunk 2 above)
Comment (addjuarez) [reply specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json:830]: Added

File: specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json (hunk 8 above)
Comment (addjuarez) [reply specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/CommonDefinitions.json:894]: Added

**Comment by addjuarez** [comment]
Not sure about the breaking changes part. They are referencing code unrelated to my changes.

**Comment by mikocot** [comment]
could you please let us know why this one was closed without merging? Was it merged somewhere else or all that effort was rejected?

**Pull request closed without merging**